demos/smoke/android/build-and-install

```diff
@@ -19,7 +19,30 @@ generate_local_properties() {
 	echo "ndk.dir=${ANDROID_NDK_HOME}" >> local.properties
 }
 
-glslang=$(realpath ../../../external/glslang/build/install/bin/glslangValidator)
+# courtesy of:
+# https://superuser.com/questions/205127/how-to-retrieve-the-absolute-path-of-an-arbitrary-file-from-the-os-x
+#
+function abspath() { 
+    pushd . > /dev/null
+    if [ -d "$1" ]; then
+        cd "$1";dirs -l +0
+    else
+        cd "`dirname \"$1\"`"
+        cur_dir=`dirs -l +0`
+        if [ "$cur_dir" == "/" ]; then
+            echo "$cur_dir`basename \"$1\"`"
+        else
+            echo "$cur_dir/`basename \"$1\"`"
+        fi
+    fi
+    popd > /dev/null
+}
+
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    glslang=$(abspath ../../../external/glslang/build/install/bin/glslangValidator)
```

Comment:
realpath is available for macOS via homebrew, so I don't think you need this.  We can add it to the upstream OSX for Android section of BUILD.md.